test/functional/wallet_balance.py

```diff
@@ -257,5 +259,23 @@ def test_balances(*, fee_node_1=0):
         assert_equal(self.nodes[0].getbalance(minconf=0), total_amount + 1)  # The reorg recovered our fee of 1 coin
 
 
+        def test_lastprocessedblock_json_field():
+            """
+            Tests the lastprocessedblock-object by comparing
+            the hashes & heights between generated blocks.
+            """
+            self.log.info("Test getbalances returns expected lastprocessedblock json object")
+            self.nodes[0].generatetoaddress(2, self.nodes[0].get_deterministic_priv_key().address)
+            balances = self.nodes[0].getbalances()
+            assert_greater_than(balances['mine']['immature'], 0)
+            prev_hash = self.nodes[0].getbestblockhash()
+            prev_height = self.nodes[0].getblock(prev_hash)['height']
+            self.nodes[0].generatetoaddress(5, self.nodes[0].get_deterministic_priv_key().address)
+            lastblock = self.nodes[0].getbalances()['lastprocessedblock']
+            assert_is_hash_string(lastblock['hash'])
+            assert_equal((prev_hash == lastblock['hash']), False)
+            assert_greater_than(lastblock['height'], prev_height)
+        test_lastprocessedblock_json_field()
```

Comment:
nit: would add an empty line in-between here, to mark the end of the function visually (same as for `test_balances()` above).